File: src/main/java/red/tetracube/iot/smartigloohub/messagingbroker/payloads/MessagingBrokerEventRequest.java
package red.tetracube.iot.smartigloohub.messagingbroker.payloads;

import com.fasterxml.jackson.annotation.JsonProperty;

import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;

public record MessagingBrokerEventRequest(@JsonProperty("username") @NotEmpty @NotNull String username,
                                          @JsonProperty("clientid") @NotNull @NotEmpty String clientId,
                                          @JsonProperty("action") @NotEmpty @NotNull String action
) {

}

File: src/main/kotlin/red/tetracube/iot/smartigloohub/messagingbroker/api/payloads/MessagingBrokerEventRequest.kt
package red.tetracube.iot.smartigloohub.messagingbroker.api.payloads

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty
import java.util.UUID
import javax.validation.constraints.NotEmpty
import javax.validation.constraints.NotNull

data class MessagingBrokerEventRequest @JsonCreator constructor(
    @JsonProperty("username")
    @field:NotEmpty
    @field:NotNull
    val username: String? = null,

    @JsonProperty("clientid")
    val clientId: @NotNull UUID? = null,

    @JsonProperty("action")
    @NotEmpty @NotNull
    val action: String? = null
)